agimus_controller_ros/hpp_agimus_controller.py

```diff
@@ -152,33 +155,37 @@ def send(self, sensor_msg, u, k):
         self.control_publisher.publish(self.control_msg)
 
     def update_predictions_and_refs_arrays(self):
-        if self.mpc_iter < self.mpc.whole_x_plan.shape[0]:
+        if self.mpc_iter < self.nb_mpc_iter_to_save:
             xs = self.mpc.ocp.solver.xs
             us = self.mpc.ocp.solver.us
             x_ref, p_ref, u_ref = self.mpc.get_reference()
             self.fill_predictions_and_refs_arrays(
                 self.mpc_iter, xs, us, x_ref, p_ref, u_ref
             )
-        if self.mpc_iter == self.mpc.whole_x_plan.shape[0]:
-            np.save("mpc_xs.npy", self.mpc_xs)
-            np.save("mpc_us.npy", self.mpc_us)
-            np.save("state_refs.npy", self.state_refs)
-            np.save("translation_refs.npy", self.translation_refs)
-            np.save("control_refs.npy", self.control_refs)
-
-    def create_predictions_and_refs_arrays(self, whole_traj_T):
-        self.mpc_xs = np.zeros([whole_traj_T, self.horizon_size, 2 * self.rmodel.nq])
-        self.mpc_us = np.zeros([whole_traj_T, self.horizon_size - 1, self.rmodel.nq])
-        self.state_refs = np.zeros([whole_traj_T, 2 * self.rmodel.nq])
-        self.translation_refs = np.zeros([whole_traj_T, 3])
-        self.control_refs = np.zeros([whole_traj_T, self.rmodel.nq])
+        if self.mpc_iter == self.nb_mpc_iter_to_save:
+            np.save("mpc_data.npy", self.mpc_data)
```

Comment:
this is not a good thing to add in a controller.
Please put that in a demo file.
Online we would export the data someway else if possible.

Reply:
how would you suggest to do it online ?

Reply:
Streaming in topics